volumina/utility/preferences.py

```diff
@@ -77,37 +77,50 @@ def setmany(self, *args: Tuple[str, str, Any]) -> None:
 
     def read(self):
         try:
-            with self._lock, open(self.path) as f:
-                return json.load(f)
+            with self._lock:
+                return json.loads(self.path.read_text())
         except FileNotFoundError:
             return {}
+        except Exception:
+            logger.exception("Failed to read preferences from %s", self.path)
+            return {}
 
-    def write(self, data):
-        with self._lock:
-            os.makedirs(self.path.parent, exist_ok=True)
-            with open(self.path, "w") as f:
-                json.dump(data, f, indent=2, sort_keys=True)
+    def write(self, data) -> bool:
+        try:
+            with self._lock:
+                os.makedirs(self.path.parent, exist_ok=True)
+                # Touch the file only after successful serialization
+                # in order to prevent existing file corruption.
+                self.path.write_text(json.dumps(data, indent=2, sort_keys=True))
+            return True
+        except Exception:
+            logger.exception("Failed to write preferences to %s", self.path)
+            self.path.unlink()
```

Comment:
only small issue, when deleting the file, we will not be able to investigate the cause easily. Maybe rename to `prefs.broken` or something?

Reply:
How about just leaving the file as-is? This can fail only if there is a serialization error (then we should indeed just leave the filesystem alone), or an I/O error (this should be rare, and there is no point in doing more I/O anyway).